app/models/ability.rb

```diff
@@ -53,19 +53,18 @@ def initialize(user, team_member = nil)
     end
 
     can :delete_membership, Team, Team do |team|
-      team.owners.include? user and Ability.num_owners(team, team_member) > 0
+      team.owners.include? user and Ability.number_of_owners_after_delete(team, team_member) > 0
     end
 
     can :delete_membership, Team, Team do |team|
-      Ability.num_owners(team, team_member) > 0 and Integer(id) == Integer(team_member)
+      Integer(id) == team_member and Ability.number_of_owners_after_delete(team, team_member) > 0 
```

Comment:
I think you accidentally removed the Integer conversion from `team_member` (where it ist necessary) and left it on `id` (where it can be omitted).